File: openclaw_republic/server/schemas.py
"""REST API 响应模型。"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class TaskSummary(BaseModel):
    task_id: str = Field(description="任务 ID")
    petition: str = Field(description="原始选民请愿内容")
    status: str = Field(description="任务执行状态（pending/running/completed/failed）")
    bill_state: str = Field(description="法案生命周期状态")
    created_at: datetime = Field(description="创建时间")


class TaskListResponse(BaseModel):
    total: int = Field(default=0, description="总数（占位符，如果未做 count 则可能不准确）")
    offset: int = Field(description="偏移量")
    limit: int = Field(description="单页限制")
    tasks: list[TaskSummary] = Field(description="任务列表")


class ActResponse(BaseModel):
    task_id: str = Field(description="任务 ID")
    act: dict[str, Any] = Field(description="完整的法案数据 (JSON)")
    created_at: datetime = Field(description="生成时间")


class DebateRound(BaseModel):
    round_number: int = Field(description="轮次")
    radical_statement: str = Field(description="激进派发言", default="")
    conservative_statement: str = Field(description="保守派发言", default="")
    conflict_score: float = Field(description="该轮的冲突分", default=0.0)


class DebateResponse(BaseModel):
    task_id: str = Field(description="任务 ID")
    rounds: list[DebateRound] = Field(default_factory=list, description="辩论发言记录回合")
    conflict_score_curve: list[float] = Field(default_factory=list, description="每个事件节点的分歧度曲线")


class VerdictResponse(BaseModel):
    task_id: str = Field(description="任务 ID")
    constitutional: bool = Field(description="是否合宪")
    ruling: str = Field(description="判决摘要")
    evidence: list[str] = Field(default_factory=list, description="判决证据")
    created_at: datetime = Field(description="判决产生时间")

File: backend/src/server/schemas.ts
/**
 * REST API 请求/响应 Zod Schema — 翻译自 Python schemas.py + routes.py 的 Pydantic 模型。
 */

import { z } from 'zod';

// ─── 请求模型 ───

export const PetitionRequestSchema = z.object({
  prompt: z.string().trim().min(1, '选民请愿内容不能为空'),
});
export type PetitionRequest = z.infer<typeof PetitionRequestSchema>;

// ─── 响应模型 ───

export const PetitionResponseSchema = z.object({
  task_id: z.string(),
  status: z.string(),
  message: z.string(),
});
export type PetitionResponse = z.infer<typeof PetitionResponseSchema>;

export const TaskStatusResponseSchema = z.object({
  task_id: z.string(),
  petition: z.string(),
  status: z.string(),
  bill_state: z.string(),
  result: z.string().nullable().default(null),
  created_at: z.string(),
  updated_at: z.string(),
});
export type TaskStatusResponse = z.infer<typeof TaskStatusResponseSchema>;

export const TaskSummarySchema = z.object({
  task_id: z.string(),
  petition: z.string(),
  status: z.string(),
  bill_state: z.string(),
  created_at: z.string(),
});
export type TaskSummary = z.infer<typeof TaskSummarySchema>;

export const TaskListResponseSchema = z.object({
  total: z.number().int().min(0).default(0),
  offset: z.number().int().min(0),
  limit: z.number().int().min(1),
  tasks: z.array(TaskSummarySchema),
});
export type TaskListResponse = z.infer<typeof TaskListResponseSchema>;

export const ActResponseSchema = z.object({
  task_id: z.string(),
  act: z.record(z.string(), z.any()),
  created_at: z.string(),
});
export type ActResponse = z.infer<typeof ActResponseSchema>;

export const DebateRoundSchema = z.object({
  round_number: z.number().int().min(1),
  radical_statement: z.string().default(''),
  conservative_statement: z.string().default(''),
  conflict_score: z.number().default(0.0),
  speaker_intervention: z.string().optional(),
});
export type DebateRound = z.infer<typeof DebateRoundSchema>;

export const DebateResponseSchema = z.object({
  task_id: z.string(),
  rounds: z.array(DebateRoundSchema).default([]),
  conflict_score_curve: z.array(z.number()).default([]),
  token_events: z.array(z.record(z.string(), z.any())).optional(),
});
export type DebateResponse = z.infer<typeof DebateResponseSchema>;

export const VerdictResponseSchema = z.object({
  task_id: z.string(),
  constitutional: z.boolean(),
  ruling: z.string(),
  evidence: z.array(z.string()).default([]),
  created_at: z.string(),
});
export type VerdictResponse = z.infer<typeof VerdictResponseSchema>;
